main.py

```diff
@@ -16,6 +16,12 @@
 def main(config):
     # Get current directory and set it as root log dir for Logger
     cwd = os.getcwd()
+
+    # TODO: fix race condition in a more elegant way
+    cwd += "/%08x" % random.getrandbits(32)
+    os.mkdir(cwd)
+    os.chdir(cwd)
+
     config.logger.logdir.root = cwd
     print(f"\nLogging directory of this run:  {cwd}\n")
```

Comment:
Since this is pretty low level, I think we should put it outside as a method in utils. What do you think?

Reply:
Moved to `utils.common` (with docstring describing what it's trying to solve).

Reply:
Nice thanks!